fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/KeySpaceDirectory.java

```diff
@@ -838,6 +839,7 @@ public enum KeyType {
 
         // Function that tests a value to see if it is of this type
         @Nonnull
+        @SpotBugsSuppressWarnings("SE_BAD_FIELD")
```

Comment:
There is something definitely weird about this check failure. I'm pretty sure enums don't even serialize in a way that requires serializing the fields, so I think some wires got crossed. But I'm definitely fine with taking this suppression to avoid blocking this PR